### PR TITLE
Fix: Cursor disappears after moving parent

### PR DESCRIPTION
**Pull request by incora-dev**
#276

### CODE REVIEW

File: src/reducers/existingThoughtMove.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default (state, { oldPath, newPath, offset }) => {
   const sameContext = equalArrays(oldContext, newContext)
   const oldThought = getThought(value, thoughtIndex)
   const newThought = removeDuplicatedContext(moveThought(oldThought, oldContext, newContext, oldRank, newRank), newContext)
-  const editing = equalPath(state.cursor, oldPath)
+  const editing = subsetThoughts(state.cursor, oldPath)
```
Comment (raineorshine) [inline src/reducers/existingThoughtMove.js:43]: Let's rename this to `isEditingPath` or `isPathInCursor`. The variable name should change to match the new semantics.

File: src/reducers/existingThoughtMove.js [hunk 2]
```diff
@@ -180,11 +180,24 @@ export default (state, { oldPath, newPath, offset }) => {
     }
   })
 
+  /**
+   * Keep cursor position on child element after moving parent
+   */
+  const generateCursorNewPath = () => {
```
Comment (raineorshine) [inline src/reducers/existingThoughtMove.js:186]: I don't think we need a function for this. You can just use local variables and expressions.

File: src/reducers/existingThoughtMove.js [hunk 3]
```diff
@@ -180,11 +180,24 @@ export default (state, { oldPath, newPath, offset }) => {
...
+    if (editing) {
+      const children = (state.cursor || []).slice(oldPath.length)
+      return [...newPath, ...children]
```
Comment (raineorshine) [inline src/reducers/existingThoughtMove.js:189]: Prefer `newPath.concat(cursorDescendantPath)`.

File: src/reducers/existingThoughtMove.js [hunk 4]
```diff
@@ -180,11 +180,24 @@ export default (state, { oldPath, newPath, offset }) => {
     }
   })
 
+  /**
+   * Keep cursor position on child element after moving parent
+   */
+  const generateCursorNewPath = () => {
+    if (editing) {
+      const children = (state.cursor || []).slice(oldPath.length)
```
Comment (raineorshine) [inline src/reducers/existingThoughtMove.js:188]: Let's call this `cursorDescendantPath`. Children refer to the subthoughts beneath a parent thought. Here rather you are getting the slice of the path that descends from the cursor.

File: src/reducers/existingThoughtMove.js (hunk 1 above)
Comment (incora-dev) [reply src/reducers/existingThoughtMove.js]: changed to isPathInCursor

**Comment by raineorshine** [comment]
Looks good, thanks!

**Pull request merged**